internal/controller/http/discovery.go

```diff
@@ -45,13 +46,15 @@ func (c *RestController) Discovery(e echo.Context) error {
 
 	driver := container.ProtocolDriverFrom(c.dic.Get)
 
-	correlationId := correlation.IdFromContext(request.Context())
+	// Use correlation id as request id since there is no request body
+	requestId := correlation.IdFromContext(request.Context())
```

Comment:
request id should be parsed from request body to see if there is any BaseRequest
https://github.com/edgexfoundry/go-mod-core-contracts/blob/f3e80aec5babd0e5b407fa321d570236bc4322f2/dtos/common/base.go#L15